vscode-trace-webviews/src/trace-viewer/vscode-trace-viewer-container.tsx

```diff
@@ -78,23 +82,25 @@ class TraceViewerContainer extends React.Component<{}, VscodeAppState>  {
           const message = event.data; // The JSON data our extension sent
           switch (message.command) {
           case VSCODE_MESSAGES.SET_EXPERIMENT:
-              this.doHandleExperimentSetSignal(convertSignalExperiment(JSONBig.parse(message.data)));
+              this.doHandleExperimentSetSignal(convertSignalExperiment(JSONBig.parse(message.data)), false);
               break;
           case VSCODE_MESSAGES.SET_TSP_CLIENT:
               this.setState({tspClientProvider: new TspClientProvider(message.data, this._signalHandler)}, () => {
                   if (message.experiment) {
-                      this.doHandleExperimentSetSignal(convertSignalExperiment(JSONBig.parse(message.experiment)));
+                      this.doHandleExperimentSetSignal(convertSignalExperiment(JSONBig.parse(message.experiment)), true);
                   }
               });
               break;
           case VSCODE_MESSAGES.ADD_OUTPUT:
               // FIXME: JSONBig.parse() create bigint if numbers are small
               // Not an issue right now for output descriptors.
-              const descriptor = JSONBig.parse(message.data);
-              this.setState({outputs: [...this.state.outputs, descriptor] });
+              if (message && message.data) {
```

Comment:
And in several of the handlers below

```suggestion
              if (message?.data) {
```